test/test_torch.py

```diff
@@ -5611,13 +5621,57 @@ def test_grad_scaling_unscale(self, device, dtype):
                     self.assertEqual(grad, torch.ones_like(grad), rtol=1e-5, atol=1e-7)
 
         # When passing lists with mismatched dtypes to a raw
-        # _amp_foreach_non_finite_check_and_unscale_ call,
+        # _amp_foreach_non_finite_check_and_unscale_ call on CUDA,
         # it's expected to fall back to single-tensor TensorIterator kernel.
         grads = [g.clone(), g.to(dtype=torch.float16)]
         torch._amp_foreach_non_finite_check_and_unscale_(grads, found_inf, inv_scale)
         for grad in grads:
             self.assertEqual(grad, torch.ones_like(grad), rtol=1e-5, atol=1e-7)
 
+        # Passing lists with mismatched devices to a raw
```

Comment:
Diff for this test too.

Reply:
Here is the diff [test.txt](https://github.com/pytorch/pytorch/files/13988818/test.txt)